common/custom_claims.c

```diff
@@ -0,0 +1,289 @@
+// Copyright (c) Open Enclave SDK contributors.
+// Licensed under the MIT License.
+
+#include <openenclave/attestation/custom_claims.h>
+#include <openenclave/internal/raise.h>
+#include <openenclave/internal/safecrt.h>
+#include <openenclave/internal/safemath.h>
+
+#include "common.h"
+
+/**
+ * Serialized buffer will have a header with version and number of
+ * claims
+ */
+typedef struct _oe_custom_claims_header
+{
+    uint64_t version;
+    uint64_t num_claims;
+} oe_custom_claims_header_t;
+
+/**
+ * Serialized entry for custom claims. Each entry will have the name and value
+ * sizes and then the contents of the name and value respectively.
+ */
+typedef struct _oe_custom_claims_entry
+{
+    uint64_t name_size;
+    uint64_t value_size;
+    uint8_t name[];
+    // name_size bytes follow.
+    // value_size_bytes follow.
+} oe_custom_claims_entry_t;
+
+oe_result_t oe_free_serialized_custom_claims(uint8_t* custom_claims_buffer)
+{
+    if (!custom_claims_buffer)
+        return OE_OK;
```

Comment:
These lines should not be needed. oe_free(), like the POSIX free(), ought to be safe to pass a null pointer to.